src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -56,6 +66,15 @@ protected override async Task OnMessageReceivedAsync(INetworkPeer peer, Incoming
                 case GetProvenHeadersPayload getHeaders:
                     await this.ProcessGetHeadersAsync(peer, getHeaders).ConfigureAwait(false);
                     break;
+
+                case HeadersPayload headers:
+                    await this.ProcessLegacyHeadersAsync(peer, headers.Headers).ConfigureAwait(false);
```

Comment:
Wouldn't call them legacy since on BTW network they are not outdated.
I'd probably rename it to process headers and for PH rename to ProcessProvenHeaders

Reply:
@noescape00 the main problem with renaming this stuff is that `ProcessGetHeadersAsync` is a base class defined in ConsensusManager and renaming it to ProcessProvenHeaderGetHeadersAsync is wrong.
This is why I needed a way to refer to the old way to handle headers, and legacy seemed appropriate to me, but if you have other suggestions let me know.
Actually I can't refactor the way you want.